engines/sherlock/decompress.cpp

```diff
@@ -0,0 +1,80 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "sherlock/decompress.h"
+
+namespace Sherlock {
+
+/**
+ * Decompresses an LZW compressed resource. If no outSize is specified, it will
+ * decompress the entire resource. If, however, an explicit size is specified,
+ * then it means we're already within a resource, and only want to decompress
+ * part of it.
+ */
+Common::SeekableReadStream *decompressLZ(Common::SeekableReadStream &source, int32 outSize) {
+	if (outSize == -1) {
+		source.seek(4);
+		outSize = source.readSint32LE();
+	}
+
+	byte lzWindow[4096];
+	uint16 lzWindowPos;
+	uint16 cmd;
+
+	byte *outBuffer = new byte[outSize];
```

Comment:
On 2015-05-16 01:38, Paul Gilbert wrote:
>> + * Decompresses an LZW compressed resource. If no outSize is specified, it will
>> + * decompress the entire resource. If, however, an explicit size is specified,
>> + * then it means we're already within a resource, and only want to decompress
>> + * part of it.
>> + */
>> +Common::SeekableReadStream *decompressLZ(Common::SeekableReadStream &source, int32 outSize) {
>> +	if (outSize == -1) {
>> +		source.seek(4);
>> +		outSize = source.readSint32LE();
>> +	}
>> +
>> +	byte lzWindow[4096];
>> +	uint16 lzWindowPos;
>> +	uint16 cmd;
>> +
>> +	byte *outBuffer = new byte[outSize];
>
> I'm curious about this "compressed" version you mention? Is it a different release than the one we've used previously?
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/scummvm/scummvm/pull/595/files#r30454320
>

I can't find my comment in the pull request anymore so I hope replying 
to the email works.

What I meant by "compressed version" is the "smaller size, slower play" 
option some versions offered on installation. My impression was that 
it's the same as the "larger size, faster play" version, except that the 
data files are uncompressed on the fly while playing the game.

But it's hard for me to tell if they're the same, because I get the 
impression that they're unpacked from two different archive files on the CD.